refactor(markdown): Change markdown syntax to be more intuitive

anoadragon453 opened the pull request:
As suggested by #3404, the current markdown syntax is unintuitive, and thus we have agreed on switching to @SkyzohKey's implementation.

zetok commented:
`Closes #3404` could be added to commit message, to automatically close the issue when PR gets merged

anoadragon453 commented:
Huh, did not know that. I'll add it in if it's not merged by the time I get back from work.

On June 26, 2016 10:00:21 AM PDT, zetok notifications@github.com wrote:

> `Closes #3404` could be added to commit message, to automatically close
> the issue when PR gets merged
> 
> ---
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/tux3/qTox/pull/3441#issuecomment-228610969

anoadragon453 commented:
Fixed errors and removed qDebug stuff.

zetok commented:
~~Would be good to also update [user manual](https://github.com/tux3/qTox/blob/4f493c5df73dffef25e4f34984fdb736f8da626c/doc/user_manual_en.md#message-styling)~~

Edit: nvm the brainfart

zetok commented:
@anoadragon453 btw, could you add a [`BREAKING CHANGE:`](https://github.com/tux3/qTox/blob/master/CONTRIBUTING.md#body) part to commit message?

zetok commented on doc/user_manual_en.md:
```diff
@@ -223,12 +223,14 @@ Similar to other messaging applications, qTox supports stylized text formatting.
 Formatting follows
 [Markdown syntax](https://daringfireball.net/projects/markdown/syntax), thus:
```
Since it's no longer markdown, those 2 lines should be removed

zetok commented on src/chatlog/chatmessage.cpp:
```diff
@@ -186,14 +183,16 @@ QString ChatMessage::detectMarkdown(const QString &str)
     QString out = str;
 
     // Create regex for certain markdown syntax
```
not a markdown anymore

zetok commented:
PR could also make some use of removing `Markdown` in settings, and in other places in code.

Though, if rename were to happen ~now, it would conflict with #3449, since it touched strings with `Markdown`.

anoadragon453 commented:
So it seems all translations have left 'Markdown' in English. Search and replacing Markdown with something like Style might break those translations.

anoadragon453 commented:
Added Breaking Change, and all instances of Markdown except in translations have been removed. See earlier comment for reasoning.

zetok commented:
> `refactor(markdown)`

Not a `refactor`, since it changes what users interact with. More a `feat`.

anoadragon453 commented:
@zetok updated.

zetok commented:
@anoadragon453 um, now PR doesn't compile?

anoadragon453 commented:
Alright! Fixed: 
- Bug in that _not_ showing formatting characters actually works
- Multi-line code formatting doesn't insert random newlines
- Settings shows 'Show Formatting Characters' by default

Rebased and yes, it compiles :)

anoadragon453 commented:
@zetok Merge?

zetok commented:
PR needs to be rebased, since there are now some conflicts.

anoadragon453 committed:
feat(textstyle): Change markdown syntax to be more intuitive

BREAKING CHANGE: Current markdown syntax is unintuitive, and thus we have agreed on switching to @SkyzohKey's implementation. Replaces all instances of Markdown with Text Styling. Closes #3404.

anoadragon453 commented:
@zetok Rebased.
